lib/cloudant.js

```diff
@@ -773,22 +768,16 @@ Cloudant.prototype.updateAll = function updateAll(model, where,
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
-
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) {
       return cb(null, {count: 0});
     }
-
-    data = getPlainJSONData.call(self, model, data);
-
-    for (var i = 0; i < docs.length; i++) {
-      _.merge(docs[i], data);
-    }
-    debug('Cloudant.prototype.updateAll bulk docs: %j', docs);
-    mo.db.bulk({docs: docs}, function(err, result) {
-      if (result) result.count = docs.length;
-      cb(err, result);
+    async.each(docs, function(doc, cb) {
```

Comment:
hmm, this solution looks good, while my concern is here we have to sacrifice the time we benefit from only talking to db once by `bulk`